src/mpd.c

```diff
@@ -3586,12 +3595,12 @@ output_get_cb(struct player_speaker_info *spk, void *arg)
   struct output_get_param *param = arg;
 
   if (!param->output
-      && param->shortid == (unsigned short) spk->id)
+      && param->shortid == param->curid++)
```

Comment:
Avoid testing and assigning on the same line, it makes it harder to read. Put the assignment on a separate line.